warning C4800: 'unsigned int': forcing value to bool 'true' or 'false' (performance warning)

I'm using VS2015/MSVC14 and the current version (a8f4dd9) on the dev branch.

I'm getting the **warning C4800: 'unsigned int': forcing value to bool 'true' or 'false' (performance warning)**.

This comes from:

https://github.com/fnc12/sqlite_orm/blob/a8f4dd91f130ea9b223d1802b5dc5172dce33f1f/include/sqlite_orm/sqlite_orm.h#L5953

The following code fixes the warning:

```cpp
(dbColumnInfo.dflt_value.length() > 0) == (storageColumnInfo.dflt_value.length() > 0) &&
```

Should I file a pull request to fix it?

Thanks for the great library!

___

## Details

My code looks the following:

```cpp
#include <iostream>
#include "sqlite_orm.h"

struct Circle
{
	int id;
	std::string name;
	double diameter;
};

int main()
{
	using namespace sqlite_orm;
	auto storage = make_storage(":memory:",
		make_table("circles",
			make_column("id", &Circle::id, autoincrement(), primary_key()),
			make_column("name", &Circle::name),
			make_column("diameter", &Circle::diameter)
		)
	);
	storage.sync_schema();
	Circle c1{-1, "t1", 13.37};
	c1.id = storage.insert(c1);
	std::cout << "inserted ID: " << c1.id << std::endl;
	Circle c2 { -1, "t2", 42.17 };
	c2.id = storage.insert(c2);
	std::cout << "inserted ID: " << c2.id << std::endl;
	Circle c2_ = storage.get<Circle>(c2.id);
	std::cout << "circle: id=" << c2_.id << ", name=" << c2_.name << ", diameter=" << c2_.diameter << std::endl;
	return 0;
}
```

I'm getting the following warning:

```
1>------ Build started: Project: SQLiteORM, Configuration: Debug Win32 ------
1>  SQLiteORM.cpp
1>  sqliteorm\sqliteorm\sqlite_orm.h(5953): warning C4800: 'unsigned int': forcing value to bool 'true' or 'false' (performance warning)
1>  sqliteorm\sqliteorm\sqlite_orm.h(5929): note: while compiling class template member function 'bool sqlite_orm::internal::storage_impl<sqlite_orm::internal::table_t<Circle,sqlite_orm::internal::column_t<Circle,int,const int &(__thiscall Circle::* )(void) const,void (__thiscall Circle::* )(int),sqlite_orm::constraints::autoincrement_t,sqlite_orm::constraints::primary_key_t<>>,sqlite_orm::internal::column_t<Circle,std::string,const std::string &(__thiscall Circle::* )(void) const,void (__thiscall Circle::* )(std::string)>,sqlite_orm::internal::column_t<Circle,double,const double &(__thiscall Circle::* )(void) const,void (__thiscall Circle::* )(double)>>>::get_remove_add_columns(std::vector<sqlite_orm::table_info *,std::allocator<_Ty>> &,std::vector<sqlite_orm::table_info,std::allocator<sqlite_orm::table_info>> &,std::vector<sqlite_orm::table_info,std::allocator<sqlite_orm::table_info>> &)'
1>          with
1>          [
1>              _Ty=sqlite_orm::table_info *
1>          ]
1>  sqliteorm\sqliteorm\sqlite_orm.h(9424): note: see reference to function template instantiation 'bool sqlite_orm::internal::storage_impl<sqlite_orm::internal::table_t<Circle,sqlite_orm::internal::column_t<Circle,int,const int &(__thiscall Circle::* )(void) const,void (__thiscall Circle::* )(int),sqlite_orm::constraints::autoincrement_t,sqlite_orm::constraints::primary_key_t<>>,sqlite_orm::internal::column_t<Circle,std::string,const std::string &(__thiscall Circle::* )(void) const,void (__thiscall Circle::* )(std::string)>,sqlite_orm::internal::column_t<Circle,double,const double &(__thiscall Circle::* )(void) const,void (__thiscall Circle::* )(double)>>>::get_remove_add_columns(std::vector<sqlite_orm::table_info *,std::allocator<_Ty>> &,std::vector<sqlite_orm::table_info,std::allocator<sqlite_orm::table_info>> &,std::vector<sqlite_orm::table_info,std::allocator<sqlite_orm::table_info>> &)' being compiled
1>          with
1>          [
1>              _Ty=sqlite_orm::table_info *
1>          ]
1>  sqliteorm\sqliteorm\sqlite_orm.h(6708): note: see reference to class template instantiation 'sqlite_orm::internal::storage_impl<sqlite_orm::internal::table_t<Circle,sqlite_orm::internal::column_t<Circle,int,const int &(__thiscall Circle::* )(void) const,void (__thiscall Circle::* )(int),sqlite_orm::constraints::autoincrement_t,sqlite_orm::constraints::primary_key_t<>>,sqlite_orm::internal::column_t<Circle,std::string,const std::string &(__thiscall Circle::* )(void) const,void (__thiscall Circle::* )(std::string)>,sqlite_orm::internal::column_t<Circle,double,const double &(__thiscall Circle::* )(void) const,void (__thiscall Circle::* )(double)>>>' being compiled
1>  sqliteorm\sqliteorm\sqliteorm.cpp(24): note: see reference to class template instantiation 'sqlite_orm::internal::storage_t<sqlite_orm::internal::table_t<Circle,sqlite_orm::internal::column_t<Circle,int,const int &(__thiscall Circle::* )(void) const,void (__thiscall Circle::* )(int),sqlite_orm::constraints::autoincrement_t,sqlite_orm::constraints::primary_key_t<>>,sqlite_orm::internal::column_t<Circle,std::string,const std::string &(__thiscall Circle::* )(void) const,void (__thiscall Circle::* )(std::string)>,sqlite_orm::internal::column_t<Circle,double,const double &(__thiscall Circle::* )(void) const,void (__thiscall Circle::* )(double)>>>' being compiled
1>  SQLiteORM.vcxproj -> SQLiteORM\Debug\SQLiteORM.exe
1>  SQLiteORM.vcxproj -> SQLiteORM\Debug\SQLiteORM.pdb (Full PDB)
```

This warning comes from:

https://github.com/fnc12/sqlite_orm/blob/a8f4dd91f130ea9b223d1802b5dc5172dce33f1f/include/sqlite_orm/sqlite_orm.h#L5953

The following code fixes the warning:

```cpp
(dbColumnInfo.dflt_value.length() > 0) == (storageColumnInfo.dflt_value.length() > 0) &&
```

--- a/dev/storage_impl.h
+++ b/dev/storage_impl.h
@@ -395,7 +395,7 @@ namespace sqlite_orm {
                             auto columnsAreEqual = dbColumnInfo.name == storageColumnInfo.name &&
                             *dbColumnInfoType == *storageColumnInfoType &&
                             dbColumnInfo.notnull == storageColumnInfo.notnull &&
-                            bool(dbColumnInfo.dflt_value.length()) == bool(storageColumnInfo.dflt_value.length()) &&
+                            (dbColumnInfo.dflt_value.length() > 0) == (storageColumnInfo.dflt_value.length() > 0) &&
                             dbColumnInfo.pk == storageColumnInfo.pk;
                             if(!columnsAreEqual){
                                 notEqual = true;

--- a/include/sqlite_orm/sqlite_orm.h
+++ b/include/sqlite_orm/sqlite_orm.h
@@ -5950,7 +5950,7 @@ namespace sqlite_orm {
                             auto columnsAreEqual = dbColumnInfo.name == storageColumnInfo.name &&
                             *dbColumnInfoType == *storageColumnInfoType &&
                             dbColumnInfo.notnull == storageColumnInfo.notnull &&
-                            bool(dbColumnInfo.dflt_value.length()) == bool(storageColumnInfo.dflt_value.length()) &&
+                            (dbColumnInfo.dflt_value.length() > 0) == (storageColumnInfo.dflt_value.length() > 0) &&
                             dbColumnInfo.pk == storageColumnInfo.pk;
                             if(!columnsAreEqual){
                                 notEqual = true;

